geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/SqlHandler.java

```diff
@@ -152,6 +156,40 @@ PdxInstance executeReadStatement(PreparedStatement statement, List<ColumnValue>
     return pdxInstance;
   }
 
+  /**
+   * @throws SQLException if the column value get fails
+   */
+  private void writeField(PdxInstanceFactory factory, ResultSet resultSet, int columnIndex,
+      String fieldName, FieldType fieldType) throws SQLException {
+    switch (fieldType) {
+      case STRING:
+        factory.writeString(fieldName, resultSet.getString(columnIndex));
+        break;
+      default:
+        factory.writeObject(fieldName, resultSet.getObject(columnIndex));
+    }
+  }
+
+  private FieldType getFieldType(Region region, String pdxClassName, String fieldName,
+      ResultSetMetaData metaData, int columnIndex) {
+    FieldType result = null;
+    if (pdxClassName != null) {
+      InternalCache cache = (InternalCache) region.getRegionService();
+      PdxType pdxType = cache.getPdxRegistry().getPdxTypeForField(fieldName, pdxClassName);
+      if (pdxType != null) {
+        PdxField pdxField = pdxType.getPdxField(fieldName);
+        if (pdxField != null) {
+          result = pdxField.getFieldType();
+        }
+      }
+    }
+    if (result == null) {
+      // TODO check metadata, for now make it object
```

Comment:
Do you want to file a Jira ticket for this TODO so we don't forget to make a further change to this code? We should add a chore or story or bug in Tracker for it.